Source/NSPathCell.m

```diff
@@ -0,0 +1,309 @@
+}
+
+- (void) setBackgroundColor: (NSColor *)color
+{
+  ASSIGNCOPY(_backgroundColor, color);
+}
+
++ (Class) pathComponentCellClass
+{
+  return pathComponentCellClass;
+}
+
++ (void) setPathComponentCellClass: (Class)clz
+{
+  pathComponentCellClass = clz;
+}
+
+- (NSRect)rectOfPathComponentCell:(NSPathComponentCell *)cell 
+                        withFrame:(NSRect)frame 
+                           inView:(NSView *)view
+{
+  return NSZeroRect;
+}
+
+- (NSPathComponentCell *)pathComponentCellAtPoint:(NSPoint)point 
+                                        withFrame:(NSRect)frame 
+                                           inView:(NSView *)view
+{
+  return nil;
+}
+
+- (NSPathComponentCell *) clickedPathComponentCell
+{
+  return nil;
+}
+
+- (NSArray *) pathComponentCells
+{
+  return _pathComponentCells;
+}
+
+- (void) setPathComponentCells: (NSArray *)cells
+{
+  ASSIGNCOPY(_pathComponentCells, cells);
+}
+
+- (SEL) doubleAction
+{
+  return _doubleAction;
+}
+
+- (void) setDoubleAction: (SEL)action
+{
+  _doubleAction = action;
+}
+
+- (NSURL *) URL
+{
+  return _url;
+}
+
+- (void) setURL: (NSURL *)url
+{
+  ASSIGNCOPY(_url, url);
+  [self setPathComponentCells: [NSPathCell _generateCellsForURL: url]];
+}
+
+- (id<NSPathCellDelegate>) delegate
+{
+  return _delegate;
+}
+
+- (void) setDelegate: (id<NSPathCellDelegate>)delegate
+{
+  _delegate = delegate;
+}
+
+- (void) drawInteriorWithFrame: (NSRect)frame inView: (NSView *)controlView
+{
+  NSUInteger count = [_pathComponentCells count];
+
+  [super drawInteriorWithFrame: frame
+                        inView: controlView];
+  if (count > 0)
+    {
+      NSEnumerator *en = [_pathComponentCells objectEnumerator];
+      NSPathComponentCell *cell = nil;
+      CGFloat cell_width = (frame.size.width / (CGFloat)[_pathComponentCells count]);
+      CGFloat current_x = 0.0;
+      
+      while((cell = (NSPathComponentCell *)[en nextObject]) != nil)
+        {
+          NSRect f = NSMakeRect(current_x, 0.0, cell_width, frame.size.height);
+          [cell drawInteriorWithFrame: f
+                               inView: controlView];
+          current_x += cell_width;
+        }
+    }
+}
+
+- (id) initWithCoder: (NSCoder *)coder
+{
+  if ([coder allowsKeyedCoding])
+    {
+      [self setPathStyle: NSPathStyleStandard];
+      
+      // if ([coder containsValueForKey: @"NSPathStyle"])
+        {
+          [self setPathStyle: [coder decodeIntegerForKey: @"NSPathStyle"]];
+        }
+      
+      if ([coder containsValueForKey: @"NSPathComponentCells"])
+        {
+          [self setPathComponentCells: [coder decodeObjectForKey: @"NSPathComponentCells"]];
+        }
+    }
+  else
+    {
+    }
+
+  return self;
+}
+
+@end
+
+@implementation NSPathCell (Private)
+
+// Private...
++ (NSArray *) _generateCellsForURL: (NSURL *)url
+{
+  NSMutableArray *array = [NSMutableArray arrayWithCapacity: 10];
+    
+  // Create cells
+  if (url != nil)
+    {
+      NSString *string = nil;
```

Comment:
I see no useful assignment to string. This will stay nil.
Also a better name would surely help to make clear what is going on here.

Reply:
This has been removed.